python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -437,43 +437,77 @@ def ExternalPartitionSetExecutionParams(self, request, _context):
             PartitionSetExecutionParamArgs,
         )
 
-        serialized_data = serialize_dagster_namedtuple(
-            get_partition_set_execution_param_data(
-                self._get_repo_for_origin(args.repository_origin),
-                partition_set_name=args.partition_set_name,
-                partition_names=args.partition_names,
-                instance=DagsterInstance.from_ref(args.instance_ref),
+        if args.instance_ref:
+            with DagsterInstance.from_ref(args.instance_ref) as instance:
+                serialized_data = serialize_dagster_namedtuple(
+                    get_partition_set_execution_param_data(
+                        self._get_repo_for_origin(args.repository_origin),
+                        partition_set_name=args.partition_set_name,
+                        partition_names=args.partition_names,
+                        instance=instance,
+                    )
+                )
+        else:
+            serialized_data = serialize_dagster_namedtuple(
+                get_partition_set_execution_param_data(
+                    self._get_repo_for_origin(args.repository_origin),
+                    partition_set_name=args.partition_set_name,
+                    partition_names=args.partition_names,
+                )
```

Comment:
you could do this fanciness

```
from contextlib import nullcontext
with (DagsterInstance.from_ref(args.instance_ref) if args.instance_ref else nullcontext()) as instance:
   ... # same body for both
```

After https://github.com/dagster-io/dagster/pull/12194 lands this can also try to pull an instance ref from self._instance_ref if there is no args.instance_ref